src/cpp/library/algorithm

```diff
@@ -14,725 +14,868 @@
-		FwdIt2 *last2) {
-	for (; first1 != last1; ++first1)
-		for (FwdIt2 *it = first2; it != last2; ++it)
-			if (*it == *first1)
-				return first1;
-	return last1;
-}
-
-template<class FwdIt1, class FwdIt2, class Pr>
-FwdIt1 find_first_of(FwdIt1 first1, FwdIt1 last1, FwdIt2 first2, FwdIt2 last2,
-		Pr pred) {
-	for (; first1 != last1; ++first1)
-		for (FwdIt2 it = first2; it != last2; ++it)
-			if (pred(*it, *first1))
-				return first1;
-	return last1;
-}
-
-template<class FwdIt1, class FwdIt2, class Pr>
-FwdIt1 find_first_of(FwdIt1 first1, FwdIt1 last1, FwdIt2 *first2, FwdIt2 *last2,
-		Pr pred) {
-	for (; first1 != last1; ++first1)
-		for (FwdIt2 *it = first2; it != last2; ++it)
-			if (pred(*it, *first1))
-				return first1;
-	return last1;
-}
-
-template<class FwdIt>
-FwdIt adjacent_find(FwdIt first, FwdIt last) {
-	if (first != last) {
-		FwdIt next = first;
-		++next;
-		while (next != last) {
-			if (*first == *next)
-				return first;
-			else {
-				++first;
-				++next;
-			}
-		}
-	}
-	return last;
-}
-
-template<class FwdIt, class Pr>
-FwdIt adjacent_find(FwdIt first, FwdIt last, Pr pred) {
-	if (first != last) {
-		FwdIt next = first;
-		++next;
-		while (next != last) {
-			if (pred(*first, *next))
-				return first;
-			else {
-				++first;
-				++next;
-			}
-		}
-	}
-	return last;
-}
-
-//template<class InIt, class Ty, class Dist>
-//int count(InIt first, InIt last, const Ty& val);
-
-
-template<class InputIterator, class T>
-int count(InputIterator first, InputIterator last, const T& value) {
-	int ret = 0;
-	while (first != last)
-		if (*first++ == value)
-			++ret;
-	return ret;
-}
-
-template<class InputIterator, class T>
-int count(InputIterator *first, InputIterator *last, const T& value) {
-	int ret = 0;
-	while (first != last)
-		if (*first++ == value)
-			++ret;
-	return ret;
-}
-
-template<class InputIterator, class T>
-int count_if(InputIterator first, InputIterator last, const T& pred) {
-	int ret = 0;
-	while (first != last)
-		if (pred(*first++))
-			++ret;
-	return ret;
-}
-
-template<class InIt1, class InIt2>
-pair<InIt1, InIt2> mismatch(InIt1 first1, InIt1 last1, InIt2 first2) {
-	while ((first1 != last1) && (*first1 == *first2)) {
-		++first1;
-		++first2;
-	}
-	return make_pair(first1, first2);
-}
-
-template<class InIt1, class InIt2>
-pair<InIt1, InIt2> mismatch(InIt1 first1, InIt1 last1, InIt2 *first2) {
-	while ((first1 != last1) && (*first1 == *first2)) {
-		++first1;
-		++first2;
-	}
-	return make_pair(first1, first2);
-}
-
-template<class InIt1, class InIt2, class Pr>
-pair<InIt1, InIt2> mismatch(InIt1 first1, InIt1 last1, InIt2 first2, Pr pred) {
-	while (pred(*first1, *first2)) {
-		++first1;
-		++first2;
-	}
-	return make_pair(first1, first2);
-}
-
-template<class InIt1, class InIt2>
-bool equal(InIt1 first1, InIt1 last1, InIt2 first2) {
-	while (first1 != last1) {
-		if (!(*first1 == *first2))
-			return false;
-		++first1;
-		++first2;
-	}
-	return true;
-}
+typedef bool(funcao)(int arg1, int arg2);
```

Comment:
Should we consider removing the Portuguese?

Reply:
Yes :-) This OM was written by @feliperodri in 2012 :-)

Reply:
Done.